Daemon runtime files missing from .gitignore trigger shepherd dirty-main check

## Problem

The daemon creates runtime files that are not in `.gitignore`, causing the shepherd's dirty-main check to fail:

- `.loom/daemon-loop.pid` — not gitignored (`.loom/.daemon.pid` is gitignored but that's a different file with dot prefix)
- `.loom/daemon-metrics.json` — not gitignored

When the daemon runs and spawns shepherds, `git status --porcelain` shows these as untracked (`??`), and the shepherd's `_check_main_repo_clean()` function in `loom-tools/src/loom_tools/shepherd/cli.py:173-216` treats them as uncommitted changes, blocking execution with exit code 4.

## Impact

Every shepherd spawned by the daemon immediately fails with:
```
[ERROR] Main repo has uncommitted changes that could cause test inconsistencies.
```

The shepherd retries with `--allow-dirty-main` (wasting time), or stalls entirely.

## Fix

Add to `.gitignore`:
```
.loom/daemon-loop.pid
.loom/daemon-metrics.json
```

## Files

- `.gitignore` — missing entries
- `loom-tools/src/loom_tools/daemon_v2/context.py` — creates `daemon-loop.pid`
- `loom-tools/src/loom_tools/daemon_v2/cli.py` — creates `daemon-metrics.json`
- `loom-tools/src/loom_tools/shepherd/cli.py:173-216` — dirty-main check
- `loom-tools/src/loom_tools/common/git.py:225-249` — `get_uncommitted_files()`

--- a/.gitignore
+++ b/.gitignore
@@ -39,6 +39,8 @@ Thumbs.db
 .loom/.daemon.log
 .loom/daemon.sock
 .loom/daemon-state.json
+.loom/daemon-loop.pid
+.loom/daemon-metrics.json
 .loom/loom-source-path
 .loom/[0-9][0-9]-daemon-state.json
 .loom/stuck-history.json

--- a/defaults/scripts/loom-daemon.sh
+++ b/defaults/scripts/loom-daemon.sh
@@ -83,10 +83,10 @@ fi
 
 if [[ -n "$LOOM_TOOLS" ]] && [[ -x "$LOOM_TOOLS/.venv/bin/loom-daemon" ]]; then
     # Use venv from loom-tools directory
-    exec "$LOOM_TOOLS/.venv/bin/loom-daemon" "${args[@]}"
+    exec "$LOOM_TOOLS/.venv/bin/loom-daemon" ${args[@]+"${args[@]}"}
 elif command -v loom-daemon &>/dev/null; then
     # System-installed
-    exec loom-daemon "${args[@]}"
+    exec loom-daemon ${args[@]+"${args[@]}"}
 else
     echo "[ERROR] Python daemon not available." >&2
     echo "" >&2

